libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -290,6 +301,26 @@ void AP_ICEngine::update(void)
         }
     }
 
+    // check against redline RPM
+    if (rpm_instance > 0) {
+        if (redline_rpm > 0 && rpm.get_rpm(rpm_instance-1, rpm_value)) {
+            // update the filtered RPM value
+            filtered_rpm_value =  _rpm_filter.apply(rpm_value);
+            if (filtered_rpm_value > redline_rpm) {
+                // check if RPM is over the redline value
+                if (!redline.flag) {
+                    gcs().send_text(MAV_SEVERITY_INFO, "Engine: Above redline RPM");
+                }
+                redline.flag = true;
+            } else if (filtered_rpm_value < redline_rpm * 0.9f) {
+                redline.flag = false;
+                // reset redline governor
+                redline.throttle_percentage = 0.0f;
+                redline.governor_integrator = 0.0f;
+            }
+        }
+    }
```

Comment:
Need to clear the redline flag here if the recliner rpm is <0 or else it can get stuck on if you change the pattern while the flag is high

Reply:
Added an else to set the flag false.